uaclient/contract_data_types.py

```diff
@@ -0,0 +1,572 @@
+    fields = [
+        Field("id", StringDataValue, False),
+        Field("renewalItems", data_list(RenewalItem), False),
+        Field("start", StringDataValue, False),
+        Field("end", StringDataValue, False),
+        Field("externalAccountID", ExternalID, False),
+        Field("externalAssetIDs", ExternalID, False),
+        Field("createdAt", StringDataValue, False),
+        Field("createdBy", StringDataValue, False),
+        Field("lastModified", StringDataValue, False),
+        Field("contractID", StringDataValue, False),
+        Field("status", StringDataValue, False),
+        Field("billing", BillingInfo, False),
+        Field("actionable", BoolDataValue, False),
+        Field("stripeInvoices", data_list(StripeInvoice), False),
+        Field("newContractStart", StringDataValue, False),
+    ]
+
+    def __init__(
+        self,
+        id: Optional[str],
+        renewalItems: Optional[List[RenewalItem]],
+        start: Optional[str],
+        end: Optional[str],
+        externalAccountID: Optional[ExternalID],
+        externalAssetIDs: Optional[ExternalID],
+        createdAt: Optional[str],
+        createdBy: Optional[str],
+        lastModified: Optional[str],
+        contractID: Optional[str],
+        status: Optional[str],
+        billing: Optional[BillingInfo],
+        actionable: Optional[bool],
+        stripeInvoices: Optional[List[StripeInvoice]],
+        newContractStart: Optional[str],
+    ):
+        self.id = id
+        self.renewalItems = renewalItems
+        self.start = start
+        self.end = end
+        self.externalAccountID = externalAccountID
+        self.externalAssetsIDs = externalAssetIDs
+        self.createdAt = createdAt
+        self.createdBy = createdBy
+        self.lastModified = lastModified
+        self.contractID = contractID
+        self.status = status
+        self.billing = billing
+        self.actionable = actionable
+        self.stripeInvoices = stripeInvoices
+        self.newContractStart = newContractStart
+
+
+class ContractItem(DataObject):
+    fields = [
+        Field("id", IntDataValue, False),
+        Field("contractID", StringDataValue, False),
+        Field("created", StringDataValue, False),
+        Field("lastModified", StringDataValue, False),
+        Field("effectiveFrom", StringDataValue, False),
+        Field("effectiveTo", StringDataValue, False),
+        Field("metric", StringDataValue, False),
+        Field("value", IntDataValue, False),
+        Field("reason", StringDataValue, False),
+        Field("externalIDs", ExternalID, False),
+        Field("purchaseID", StringDataValue, False),
+        Field("productListingID", StringDataValue, False),
+        Field("disabled", BoolDataValue, False),
+        Field("disabledExplanation", StringDataValue, False),
+        Field("subscriptionID", StringDataValue, False),
+        Field("period", StringDataValue, False),
+        Field("renewal", Renewal, False),
```

Comment:
I think should can drop `renewal` and all its nested content - lots of it is bordering on sensitive, and we don't use any of it anyway

Reply:
Okay. Makes sense.